scalableminds/webknossos
# task request no longer relies on the absence of zero-instance OpenAssignments

This PR is an attempt to tackle the heisenbug issue #2046 .
One possible reason for the bug might be that the old task request code relies on the absence of zero-instance OpenAssignments in the database. If their deletion is delayed by the database, they might be the source of wrong »no tasks available« messages.

### Changes
 - When requesting a new task, only consider OpenAssignments with instances > 0
 - Zero-instance OpenAssignments are no longer removed upon failed requests
 - Instead they are removed periodically by a cleanup job

### Steps to test:
- To test that the task request still works, create tasks in admin view, let users take and finish them, check if the instance counts match
- To test if this fixes the issue, deploy and check if the warning »Failed to remove any assignment for user« does not come up again in the logfiles (possibly in /var/log/)

### Issues:
- tackles #2046 (and might fix it)

------
